File: react-type-vite/src/services/api/request/quizRequest.ts
import type { QuestionRequest } from "./questionRequest";

export interface QuizRequest {
  id?: number;
  sectionId?: number;
  title: string;
  description?: string;
  duration: number; // Made required to match backend
  attemptLimit?: number;
  passingLimit?: number;
  passingScore?: number; // Changed to number from backend Double
  numberItem?: number;
  showResults?: boolean;
  isPublished?: boolean;
  startTime?: string;
  endTime?: string;
  questions?: QuestionRequest[]; // For frontend display only
  questionIds?: number[]; // IDs for many-to-many relationship
  blueprintDraft?: { cloId: number; percentage: number }[]; // Local draft before persisted quizId exists
  createdAt?: string;
  updateAt?: string;
}

File: spring-microservices/course-service/src/main/java/com/hoangphihiep/dto/request/QuizRequest.java
package com.hoangphihiep.dto.request;

import jakarta.persistence.Column;
import jakarta.validation.Valid;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.Date;
import java.util.Set;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class QuizRequest {

    private Integer id;

    @NotNull(message = "Section ID is required")
    private Integer sectionId;

    @NotBlank(message = "Quiz title is required")
    private String title;

    @Column(length = 1000)
    private String description;

    @NotNull(message = "Duration is required")
    private Integer duration;

    private Integer attemptLimit;

    private Double passingScore;

    private Integer numberItem;

    private Boolean showResults;

    private Boolean isPublished;

    private LocalDateTime startTime;

    private LocalDateTime endTime;

    @Valid
    private Set<QuestionRequest> questions; // For backward compatibility (optional)
    
    private Set<Integer> questionIds; // For many-to-many relationship

    private Date createdAt;

    private Date updateAt;
}
